### PR TITLE
remove VeleroConfigBackupProxy

### PR DESCRIPTION
**Pull request by birsanv**
Signed-off-by: Valentina Birsan <vbirsan@redhat.com>

https://github.com/open-cluster-management/backlog/issues/15738

The VeleroConfigBackupProxy spec property was used to define the namespace of the velero resources
Since we need to create our cluster backup in the same namespace with the velero resources, this information is redundant

I updated the Readme to remove the reference to this property

I did not make any changes to the cluster restore resource since Dario is working on the restore, trying to avoid conflicts if possible; we can update restore later on

### CODE REVIEW

File: controllers/schedule_controller.go [hunk 1]
```diff
@@ -66,7 +66,7 @@ type BackupScheduleReconciler struct {
 //+kubebuilder:rbac:groups=cluster.open-cluster-management.io,resources=backupschedules/status,verbs=get;update;patch
 //+kubebuilder:rbac:groups=cluster.open-cluster-management.io,resources=backupschedules/finalizers,verbs=update
 //+kubebuilder:rbac:groups=cluster.open-cluster-management.io,resources=managedclusters,verbs=get;list;watch
-//+kubebuilder:rbac:groups=cluster.open-cluster-management.io,resources=channels,verbs=get;list;watch
+//+kubebuilder:rbac:groups=apps.open-cluster-management.io,resources=channels,verbs=get;list;watch
 //+kubebuilder:rbac:groups=velero.io,resources=schedules,verbs=get;list;watch;create;update;patch
```
Comment (sahare) [inline controllers/schedule_controller.go:70]: Thanks for the fix :)

**Review by sahare — APPROVED** [review]
Looks good, Thanks!

File: controllers/schedule_controller.go (hunk 1 above)
Comment (sdminonne) [reply controllers/schedule_controller.go:70]: I would expect an update in `role.yaml` file, did you run `make manifests`?

**Review by sdminonne — APPROVED** [review]
It's OK, Guess we need to update the `role.yaml`, though. 
Thanks @birsanv !

**Comment by birsanv** [comment]
https://github.com/open-cluster-management/cluster-backup-operator/pull/30#pullrequestreview-745044932
@sdminonne the role.yaml contains the channel rbac, I did run `make manifests` 

https://github.com/open-cluster-management/cluster-backup-operator/blob/main/config/rbac/role.yaml#L12

**Comment by sdminonne** [comment]
I didn'see the file. Sorry.

**Pull request merged**